plugins/out_kinesis_streams/kinesis_api.c

```diff
@@ -97,6 +97,31 @@ static int init_put_payload(struct flb_kinesis *ctx, struct flush *buf,
     return -1;
 }
 
+/*
+ * Adjust the time_key output format according to time_key_format.
+ */
+static void get_time_format_string(char *str, size_t maxsize,
+                      const char *time_key_format, const struct tm *timeptr,
+                      size_t len, struct flb_time *tms){
+    size_t s;
+    const char *format = "%Y-%m-%dT%H:%M:%S";
```

Comment:
Wait is format supposed to be the default format? Sorry I am very confused here.

Reply:
"%Y-%m-%dT%H:%M:%S.%L" => "2022-11 ... 12:30:45.%L" => 

"%Y-%m-%dT%H:%M:%S.%L" => "%Y-%m-%dT%H:%M:%S.23923942332" => strftime